### PR TITLE
Add GH Action to run ES Linter on PRs into `main`.

### PR DESCRIPTION
**Pull request by tmeyer2115**
TEST=manual

Set ESLint Warnings to Errors and saw the PR check fail. Reverted this change and saw the PR check pass.

### CODE REVIEW

File: .github/workflows/run-linting.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+      - run: npm ci
+      - run: npm run lint
```
Comment (oshi97) [inline .github/workflows/run-linting.yml:17]: Do we want to make the workflow fail with --max-warnings? It's super sweet we get "check annotations" even without the workflow failing

File: .github/workflows/run-linting.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Run Linter
+
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (oshi97) [inline .github/workflows/run-linting.yml:6]: maybe we just run this on all pull requests?

Comment (tmeyer2115) [reply .github/workflows/run-linting.yml:6]: We could! I don't have a strong opinion either way. My thoughts for not having it on all PRs was that people may have dev branches going into feature branches. Maybe their work is still in progress and they'll get to the linting stuff later on.

File: .github/workflows/run-linting.yml (hunk 1 above)
Comment (tmeyer2115) [reply .github/workflows/run-linting.yml:17]: Sure, I can do that!

File: .github/workflows/run-linting.yml (hunk 2 above)
Comment (oshi97) [reply .github/workflows/run-linting.yml:6]: Ah ok makes sense, I was thinking about our forgetting to add our future develop branch but I guess we can just not forget

File: .github/workflows/run-linting.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Run Linter
```
[the remark this answers is not in the document]
Comment (oshi97) [reply .github/workflows/run-linting.yml:1]: Tom said we plan to add it later, we can probably say so in the PR description

[the remark this answers is not in the document]
Comment (tmeyer2115) [reply .github/workflows/run-linting.yml:1]: Eventually we will migrate it, yes.

**Review by oshi97 — APPROVED** [review]

**Pull request merged**